Source/CesiumRuntime/Private/Tests/SubLevels.spec.cpp

```diff
@@ -0,0 +1,338 @@
+      [this]() {
+        BeforeEach(EAsyncExecution::TaskGraphMainThread, [this]() {
+          pSubLevel1->SetIsTemporarilyHiddenInEditor(false);
+        });
+        BeforeEach(EAsyncExecution::TaskGraphMainThread, [this]() {
+          pSubLevel1->SetIsTemporarilyHiddenInEditor(true);
+        });
+        It("", EAsyncExecution::TaskGraphMainThread, [this]() {
+          double expectedLongitude = pGeoreference->OriginLongitude;
+          double expectedLatitude = pGeoreference->OriginLatitude;
+          double expectedHeight = pGeoreference->OriginHeight;
+
+          TestNotEqual("Longitude", expectedLongitude, 7.0);
+          TestNotEqual("Latitude", expectedLatitude, 8.0);
+          TestNotEqual("Height", expectedHeight, 9.0);
+
+          pLevelComponent1->SetOriginLongitudeLatitudeHeight(
+              FVector(7.0, 8.0, 9.0));
+          TestEqual(
+              "Longitude",
+              pGeoreference->OriginLongitude,
+              expectedLongitude);
+          TestEqual(
+              "Latitude",
+              pGeoreference->OriginLatitude,
+              expectedLatitude);
+          TestEqual("Height", pGeoreference->OriginHeight, expectedHeight);
+        });
+      });
+
+  Describe(
+      "ensures only one sub-level instance is visible in the editor at a time",
+      [this]() {
+        BeforeEach(EAsyncExecution::TaskGraphMainThread, [this]() {
+          pSubLevel1->SetIsTemporarilyHiddenInEditor(false);
+        });
+        BeforeEach(EAsyncExecution::TaskGraphMainThread, [this]() {
+          TestFalse(
+              "pSubLevel1 is hidden",
+              pSubLevel1->IsTemporarilyHiddenInEditor(true));
+          TestTrue(
+              "pSubLevel2 is hidden",
+              pSubLevel2->IsTemporarilyHiddenInEditor(true));
+
+          pSubLevel2->SetIsTemporarilyHiddenInEditor(false);
+        });
+        It("", EAsyncExecution::TaskGraphMainThread, [this]() {
+          TestTrue(
+              "pSubLevel1 is hidden",
+              pSubLevel1->IsTemporarilyHiddenInEditor(true));
+          TestFalse(
+              "pSubLevel2 is hidden",
+              pSubLevel2->IsTemporarilyHiddenInEditor(true));
+        });
+      });
+
+  Describe("activates the closest sub-level that is in range", [this]() {
+    LatentBeforeEach(
+        EAsyncExecution::TaskGraphMainThread,
+        [this](const FDoneDelegate& done) {
+          subscriptionPostPIEStarted =
+              FEditorDelegates::PostPIEStarted.AddLambda(
+                  [done](bool isSimulating) { done.Execute(); });
+          FRequestPlaySessionParams params{};
+          GEditor->RequestPlaySession(params);
+        });
+    BeforeEach(EAsyncExecution::TaskGraphMainThread, [this]() {
+      FEditorDelegates::PostPIEStarted.Remove(subscriptionPostPIEStarted);
+    });
+    LatentBeforeEach(
+        EAsyncExecution::TaskGraphMainThread,
+        [this](const FDoneDelegate& done) {
+          waitFor(done, GEditor->PlayWorld, 5.0f, [this]() {
```

Comment:
For me this (and the other play mode test) is failing because `waitFor` times out. I don't know if you can reproduce this on your machine, but if not I can take a look and debug.

Reply:
Looks like it's failing only in UE 5.2 maybe? I'll investigate.